services/shhext/service.go

```diff
@@ -108,17 +109,27 @@ func (s *Service) InitProtocol(address string, password string) error {
 		return nil
 	}
 
+	digest := sha3.Sum256([]byte(password))
+	hashedPassword := fmt.Sprintf("%x", digest)
+
 	if err := os.MkdirAll(filepath.Clean(s.dataDir), os.ModePerm); err != nil {
 		return err
 	}
-	oldPath := filepath.Join(s.dataDir, fmt.Sprintf("%x.db", address))
-	newPath := filepath.Join(s.dataDir, fmt.Sprintf("%s.db", s.installationID))
+	v0Path := filepath.Join(s.dataDir, fmt.Sprintf("%x.db", address))
+	v1Path := filepath.Join(s.dataDir, fmt.Sprintf("%s.db", s.installationID))
+	v2Path := filepath.Join(s.dataDir, fmt.Sprintf("%s.v2.db", s.installationID))
 
-	if err := chat.MigrateDBFile(oldPath, newPath, password); err != nil {
+	if err := chat.MigrateDBFileV1(v0Path, v1Path, password); err != nil {
 		return err
 	}
 
-	persistence, err := chat.NewSQLLitePersistence(newPath, password)
+	if err := chat.MigrateDBFileV2(v1Path, v2Path, password, hashedPassword); err != nil {
+		// Remove db file as created with a blank password
+		os.Remove(v1Path)
```

Comment:
Does this mean that if `MigrateDBFileV2` fails to rename `v1Path` into `v2Path` for some reason, we'll delete the source `v1Path` file? Why is that?

Reply:
basically it happened that those impacted would create a database with an empty password, so we are not able to rekey, it handles this case. Also this db `v1` can be safely erased, and we know that if it errors likely was never used.

Reply:
Maybe we should add that context in the comment?